### DISCUSSION TITLE
Decryption failing since 18 April

### DISCUSSION THREAD

**Discussion by nickwrightmaintel** [openpgpjs/openpgpjs#1749] [Q&A]
I have a Lambda function that uses OpenPGP to decrypt files. Since last Thursday (18 April) that process has failed with the following error:

{ "errorType": "Error", "errorMessage": "Error decrypting message: Decryption error", "stack": [ "Error: Error decrypting message: Decryption error", " at /opt/node_modules/openpgp/dist/node/openpgp.min.js:2:178460", " at process.processTicksAndRejections (node:internal/process/task_queues:95:5)", " at async ju.decrypt (/opt/node_modules/openpgp/dist/node/openpgp.min.js:2:291018)", " at async /opt/node_modules/openpgp/dist/node/openpgp.min.js:2:345501", " at async Promise.all (index 0)", " at async /opt/node_modules/openpgp/dist/node/openpgp.min.js:2:344837", " at async Promise.all (index 0)", " at async /opt/node_modules/openpgp/dist/node/openpgp.min.js:2:344447", " at async Promise.all (index 0)", " at async Bh.decryptSessionKeys (/opt/node_modules/openpgp/dist/node/openpgp.min.js:2:344404)" ] }

It worked fine up to this point and has continued to work fine if the Lambda function is locally debugged using VS Code. It's only when executed from AWS directly or via an EventBridge schedule that it errors.

**Comment by larabr** [comment]
Are you using the latest v5 version ([v5.11.1](https://github.com/openpgpjs/openpgpjs/releases/tag/v5.11.1))? If not, try updating and see if the problem is solved, as it patches a Node-related decryption issue with RSA.

**Reply by nickwrightmaintel** [reply]
I'm using 5.10.2, tried upgrading to latest and it gave me that. I've identified the issue was down to AWS upgrading the Node runtime from 18.v21 to 18.v26 last Thursday. I tried upgrading to v20, but it has the same bug. Node v21 is fine when running locally, but not available as an AWS runtime. I've rolled back to 18.v21 and resolved the issue. Thanks for your interest.
